editors/code/src/scopes.ts

```diff
@@ -0,0 +1,146 @@
+import * as fs from 'fs';
+import * as jsonc from 'jsonc-parser';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+export interface TextMateRule {
+    scope: string | string[];
+    settings: TextMateRuleSettings;
+}
+
+export interface TextMateRuleSettings {
+    foreground: string | undefined;
+    background: string | undefined;
+    fontStyle: string | undefined;
+}
+
+// Current theme colors
+const rules = new Map<string, TextMateRuleSettings>();
+
+export function find(scope: string): TextMateRuleSettings | undefined {
+    return rules.get(scope);
+}
+
+// Load all textmate scopes in the currently active theme
+export function load() {
+    // Remove any previous theme
+    rules.clear();
+    // Find out current color theme
+    const themeName = vscode.workspace
+        .getConfiguration('workbench')
+        .get('colorTheme');
+
+    if (typeof themeName !== 'string') {
+        // console.warn('workbench.colorTheme is', themeName)
+        return;
+    }
+    // Try to load colors from that theme
+    try {
+        loadThemeNamed(themeName);
+    } catch (e) {
+        // console.warn('failed to load theme', themeName, e)
+    }
+}
+
+function filterThemeExtensions(extension: vscode.Extension<any>): boolean {
+    return (
+        extension.extensionKind === vscode.ExtensionKind.UI &&
+        extension.packageJSON.contributes &&
+        extension.packageJSON.contributes.themes
+    );
+}
+
+// Find current theme on disk
+function loadThemeNamed(themeName: string) {
+    const themePaths = vscode.extensions.all
+        .filter(filterThemeExtensions)
+        .reduce((list, extension) => {
+            return extension.packageJSON.contributes.themes
+                .filter(
+                    (element: any) =>
+                        (element.id || element.label) === themeName
+                )
+                .map((element: any) =>
+                    path.join(extension.extensionPath, element.path)
+                )
+                .concat(list);
+        }, Array<string>());
+
+    themePaths.forEach(loadThemeFile);
+
+    const tokenColorCustomizations: [any] = [
```

Comment:
:+1:

Reply:
?

Reply:
Just happy to see you've already implemented it.